package polar;

import "directory.proto";

message SensorStatus {
	required uint32 start_index = 1;
	required uint32 stop_index = 2;
}
message SensorCalibration {

	enum Operation {
		UNKNOWN = 1;
	}
	enum Cause {
		UNKNOWN_CAUSE = 0;
		OTHER_CAUSE = 1;
	}

	required uint32 start_index = 1;
	required float value = 2;
	required Operation operation = 3;
	required Cause cause = 4;
}

message SamplesF28 {
	repeated int32 f1 = 1 [packed=true];
}

message Samples {
	required Time record_interval = 1;
	repeated int32 heartrate = 2 [packed=true];
	repeated SensorStatus heartrate_offline = 3;
	repeated int32 cadence = 4 [packed=true];
	repeated SensorStatus cadence_offline = 5;
	repeated float altitude = 6; 
	repeated float speed = 9;
	repeated float distance = 11;
	repeated SensorCalibration stride_calibration = 15;
	repeated SensorStatus altitude_offline = 18;
	repeated SamplesF28 samplesf28 = 28;
}

message Route {
	repeated int32 duration = 1 [packed=true];
	repeated double latitude = 2 [packed=true];
	repeated double longitude = 3 [packed=true];
	repeated sint32 altitude = 4 [packed=true];
	repeated int32 satellites = 5 [packed=true];
	required DateTime timestamp = 9; //? copying from bipolar code, weird that it skips a bunch of fields?
	// Actually, my ROUTE file has field 9, then field 5(which is length delimited, not an integer?), then 2 (length delimited), 3 (length delimited), 4 (length delimited), and 1 (length delimited)
}

